src/test/java/com/google/sps/WebSocketHandlerTest.java

```diff
@@ -0,0 +1,171 @@
+package com.google.sps;
+
+import com.google.sps.WebSocketHandler;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collections;
+import javax.websocket.Session;
+
+import org.junit.Test;
+import org.junit.Before;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+import static org.mockito.Mockito.*;
+
+public class WebSocketHandlerTest {
+
+  Constructor constructor;
+  Field instance;
+  Field chatRoomMap;
+  WebSocketHandler obj;
+  Map<Long, List<Session>> actualMap;
+
+  static long TEST_ROOM_ID;
+  static Session TEST_SESSION_1;
+  static Session TEST_SESSION_2;
+
+  @Before
+  public void setUp() throws NoSuchFieldException, NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
+    TEST_SESSION_1 = mock(Session.class);
+    TEST_SESSION_2 = mock(Session.class);
+    TEST_ROOM_ID = 1;
+
+    constructor = WebSocketHandler.class.getDeclaredConstructor();
+    constructor.setAccessible(true);
+
+    instance = WebSocketHandler.class.getDeclaredField("instance");
+    instance.setAccessible(true);
+
+    chatRoomMap = WebSocketHandler.class.getDeclaredField("chatRoomMap");
+    chatRoomMap.setAccessible(true);
+
+    obj = (WebSocketHandler) constructor.newInstance();
+
+    actualMap = (Map)chatRoomMap.get(obj);
+  }
+
+  @Test
+  public void getInstance_CheckNewInstanceNotNull() throws NoSuchMethodException, IllegalAccessException {
+
+    WebSocketHandler newInstance = WebSocketHandler.getInstance();
+
+    WebSocketHandler result = (WebSocketHandler)instance.get(null);
+    assertEquals(newInstance, result);
+    assertNotEquals(result, null);
+  }
+
+  @Test
+  public void getInstance_WithExistingInstance() throws InstantiationException, IllegalAccessException, InvocationTargetException {
+    instance.set(null, obj);
+
+    WebSocketHandler result = WebSocketHandler.getInstance();
+
+    assertEquals(result, obj);
+  }
+
+  @Test
+  public void addSession_WithoutExistingRoom() {
+    obj.addSession(TEST_ROOM_ID, TEST_SESSION_1);
+
+    Map<Long, List<Session>> expectedMap = new HashMap<>();
+    List<Session> expectedList = new ArrayList<>();
+    expectedList.add(TEST_SESSION_1);
+    expectedMap.put(TEST_ROOM_ID, expectedList);
+    assertEquals(actualMap, expectedMap);
+  }
+
```

Comment:
You might want to set up your tests in an assert, act, arrange style and leave empty lines between them correspondingly

Reply:
In this test the arrange section takes place in the @Before function so the only two sections left are act which is the first line and assert which is the last section. As far as I understand, setting up expected results should take place in the assert section but I may have misunderstood.

Reply:
Are there any other tests where I have not followed this structure or is it just this one?

Reply:
The structure would be clearer if you replace actualMap with a public method like getRoomList